src/Picasso_BatchedLinearAlgebra.hpp

```diff
@@ -1696,44 +1798,104 @@ operator^( const ExpressionA& a, const ExpressionB& b )
                    "value_type must be the same" );
     static_assert( ExpressionA::extent_1 == ExpressionB::extent_0,
                    "Inner extent must match" );
+    static_assert( ExpressionA::extent_1 == ExpressionA::extent_0,
+                   "matrix must be square" );
+
+    using value_type = typename ExpressionA::value_type;
+    const int m = ExpressionB::extent_0;
+    const int n = ExpressionB::extent_1;
+
+    // Compute LU decomposition.
     auto a_lu = LU( a );
+
+    // Create RHS/LHS
     typename ExpressionB::copy_type x = b;
-    KokkosBatched::SerialSolveLU<
-        KokkosBatched::Trans::NoTranspose,
-        KokkosBatched::Algo::SolveLU::Unblocked>::invoke( a_lu, x );
+
+    // Solve Ly = b for y where y = Ux
+    for ( int p = 0; p < m; ++p )
+    {
+        const int iend = m - p - 1;
+        const int jend = n;
+
+        for ( int i = 0; i < iend; ++i )
+#if defined( KOKKOS_ENABLE_PRAGMA_UNROLL )
+#pragma unroll
+#endif
+            for ( int j = 0; j < jend; ++j )
+                x( i + p + 1, j ) -= a_lu( i + p + 1, p ) * x( p, j );
```

Comment:
No need for the +/- 1 on the `i` index:
```suggestion
        const int iend = m - p;
        const int jend = n;

        for ( int i = 1; i < iend; ++i )
#if defined( KOKKOS_ENABLE_PRAGMA_UNROLL )
#pragma unroll
#endif
            for ( int j = 0; j < jend; ++j )
                x( i + p, j ) -= a_lu( i + p, p ) * x( p, j );
```

Reply:
fixed